src/H5T.c

```diff
@@ -43,6 +43,16 @@
 #include "H5VLprivate.h" /* Virtual Object Layer                     */
 #include "H5VMprivate.h" /* Vectors and arrays                       */
 
+/* Datatype conversion functions */
+#include "H5Tconv_char.h"
+#include "H5Tconv_short.h"
+#include "H5Tconv_int.h"
+#include "H5Tconv_long.h"
+#include "H5Tconv_llong.h"
+#include "H5Tconv_float.h"
+#include "H5Tconv_double.h"
+#include "H5Tconv_ldouble.h"
```

Comment:
Since all the conversion functions are currently only needed during H5T_init, I moved the prototypes into separate headers rather than H5Tconv.h so we only include them where we use them.